src/si/mod.rs

```diff
@@ -1,68 +1,318 @@
+    #[base(LuminousIntensity)]
+    unit candela: LuminousIntensity;
+
+    // derived units
+
+    #[metric_prefixes]
+    #[symbol(rad)]
+    #[alias(radians)]
+    unit radian: Angle = meter / meter;
+
+    #[metric_prefixes]
+    #[symbol(sr)]
+    #[alias(steradians)]
+    unit steradian: SolidAngle = radian^2;
+
+    #[metric_prefixes]
+    #[symbol(Hz)]
+    unit hertz: Frequency = 1 / second;
+
+    #[metric_prefixes]
+    #[symbol(N)]
+    #[alias(newtons)]
+    unit newton: Force = kilogram meter / second^2;
+
+    #[metric_prefixes]
+    #[symbol(Pa)]
+    #[alias(pascals)]
+    unit pascal: Pressure = newton / meter^2;
+
+    #[metric_prefixes]
+    #[symbol(J)]
+    #[alias(joules)]
+    unit joule: Energy = newton meter;
+
+    #[metric_prefixes]
+    #[symbol(W)]
+    #[alias(watts)]
+    unit watt: Power = joule / second;
+
+    #[metric_prefixes]
+    #[symbol(C)]
+    #[alias(coulombs)]
+    unit coulomb: ElectricCharge = ampere second;
+
+    #[metric_prefixes]
+    #[symbol(V)]
+    #[alias(volts)]
+    unit volt: Voltage = kilogram meter^2 / (second^3 ampere);
+
+    #[metric_prefixes]
+    #[symbol(F)]
+    #[alias(farads)]
+    unit farad: Capacitance = coulomb / volt;
+
+    #[metric_prefixes]
+    #[symbol(Ω)]
+    #[alias(ohms)]
+    unit ohm: ElectricResistance = volt / ampere;
+
+    #[metric_prefixes]
+    #[symbol(S)]
+    unit siemens: ElectricConductance = 1 / ohm;
+
+    #[metric_prefixes]
+    #[symbol(Wb)]
+    #[alias(webers)]
+    unit weber: MagneticFlux = volt second;
+
+    #[metric_prefixes]
+    #[symbol(T)]
+    #[alias(teslas)]
+    unit tesla: MagneticFluxDensity = weber / meter^2;
+
+    #[metric_prefixes]
+    #[symbol(H)]
+    #[alias(henrys)]
+    unit henry: Inductance = weber / ampere;
+
+    #[metric_prefixes]
+    #[symbol(lm)]
+    #[alias(lumens)]
+    unit lumen: LuminousFlux = candela steradian;
+
+    #[metric_prefixes]
+    #[symbol(lx)]
+    unit lux: Illuminance = lumen / meter^2;
+
+    #[metric_prefixes]
+    #[symbol(Bq)]
+    #[alias(becquerels)]
+    unit becquerel: Activity = 1 / second;
+
+    #[metric_prefixes]
+    #[symbol(Gy)]
+    #[alias(grays)]
+    unit gray: AbsorbedDose = joule / kilogram;
+
+    #[metric_prefixes]
+    #[symbol(Sv)]
+    #[alias(sieverts)]
+    unit sievert: EquivalentDose = joule / kilogram;
+
+    #[metric_prefixes]
+    #[symbol(kat)]
+    #[alias(katals)]
+    unit katal: CatalyticActivity = mole / second;
+
+    // SI accepted units
+    #[symbol(min)]
+    #[alias(minutes)]
+    unit minute: Time = 60 second;
+
+    #[symbol(h)]
+    #[alias(hours)]
+    unit hour: Time = 60 minute;
+
+    #[symbol(day)]
+    #[alias(days)]
+    unit day: Time = 24 hour;
+
+    #[symbol(au)]
+    #[alias(astronomicalunits)]
+    unit astronomicalunit: Length = 149_597_870_700 meter;
+
+    constant PI = 3.141592653589793;
+    //TODO(minor): Support using ° here.
+    #[symbol(deg)]
+    #[alias(degrees)]
+    unit degree: Angle = PI / 180 * radian;
```

Comment:
```suggestion
    unit degree: Angle = 180 / PI * radian;
```

Reply:
As confusing as it is, `degree = PI / 180 * radian` is the correct definition. It feels really weird because it seems like it's multiplying radian with PI (which it already contains), but a degree is less than a radian, so `PI / 180` is the correct factor.

Reply:
Gosh, never underestimate the power of covariance/contravariance to confuse. I don't know why it looked natural for the others, but jarring here. Sorry about the noise.

Reply:
No worries! When I saw your change request I was completely convinced that you were right and that the code was wrong, so I made exactly the same mistake. Thanks for looking through the definitions so thoroughly!